src/psyclone/core/signature.py

```diff
@@ -100,14 +100,17 @@ def __len__(self):
 
     # ------------------------------------------------------------------------
     def __getitem__(self, indx):
-        return self._signature[indx]
+        if isinstance(indx, slice):
+            return Signature(self._signature[indx])
+        else:
```

Comment:
Another pylint complain: unnecessary `else` (i.e. remove else, and 'outdent' the return statement

Reply:
Done.